### DISCUSSION TITLE
Definition Of Preprocessor Variables in FairRoot?

### DISCUSSION THREAD

**Discussion by kurbatov-lang** [FairRootGroup/FairRoot#1144] [General]
Dear Colleagues! Can anybody say whe I can define/redefine Preprocessor Variables in FairRoot?

I supposed that this was in CMakeLists.txt files but did not find it there.

Thank you

**Comment by dennisklein** [comment]
The preferred way is to use use the CMake command [`target_compile_definitions()`](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html).

**Comment by kurbatov-lang** [comment]
Dear colleague!

Thank you for the answer, but still it is unclear for

me : I am not well educated and let me ask you

stupid question - Where may I insert  command 
target_compile_definitions()?

As I know this command should be part of some CMakeLists.txt file.Right?

It means that I should put it in suitable file before compiling 
FairRoot?

But I would want to  know how to find setting of preprocessor variable

by some command from command line after system was compiled and 
installed?

Thank you,Vladimir

**Reply by dennisklein** [reply]
> But I would want to  know how to find setting of preprocessor variable by some command from command line after system was compiled and installed?

I am not sure I understand, what you are trying to accomplish? [Preprocessor](https://en.cppreference.com/w/cpp/preprocessor) macros are evaluated at compile-time. What effect should the definition of a preprocessor macro have on the command-line after compilation/installation? What command-line are you talking about? Do you mean in a ROOT macro or in an interactive ROOT session (aka interpreted C++)?

**Comment by kurbatov-lang** [comment]
Dear colleague!

In Root there is a class TRandom. On the other hand

geant4 is using random numbers for its functionality.

What are generators, used by it? It is not the same generators as in 
TRandom?

Thank you

**Comment by MohammadAlTurany** [comment]
Dear Vladimir,

Geant4 uses own random generators internally but the seeds you set in your macros or classes in FairRoot are passed to Geant4 through the VMC interface.

best,

Mohammad
